onnx/optimizer/optimize.h

```diff
@@ -29,6 +30,8 @@ struct Optimizer {
     passes[ftg->name] = std::move(ftg);
     std::unique_ptr<Nop> nop(new Nop());
     passes[nop->name] = std::move(nop);
+    std::unique_ptr<ShapeInference> shi(new ShapeInference());
```

Comment:
Can we use `make_unique`? It's better for exception-safety.

Reply:
`make_unique` is >=c++14, but sure you can have a fallback implementation. But IMO here doesn't need unique_ptr, the value type of `passes` should be `const OptimizePass`.